java/core/src/test/proto/com/google/protobuf/non_nested_extension.proto

```diff
@@ -45,4 +45,5 @@ message MyNonNestedExtension {}
 
 extend MessageToBeExtended {
   optional MyNonNestedExtension nonNestedExtension = 1;
+  optional string if = 2000;
```

Comment:
Protobuf have two types of extension: top level extension and non top level extension. They should have same behavior. Does non top level extension also break without append '_' ? Some languages have different generated code for them, both should be tested

The extension you tested is top level extension.

None top level extension is like:
```
message ExtensionMessage {
    extend MessageToBeExtended {
        optional ExtensionMessage ext_field = 100;
    }
}
```
For example:
https://github.com/protocolbuffers/protobuf/blob/bf32b36ab8aa9e56c031e00c0553db640d3627d4/src/google/protobuf/unittest.proto#L350

Reply:
Makes sense. I added a similar test for the nested extension in both cases. Let me know if you think we need anything more.